### PR TITLE
Bugfixes for collection create/edit and record loading

### PR DESCRIPTION
**Pull request by alexcottner**
Small PR that corrects two bugs and adds unit tests around them. This PR:
1. Corrects a known issue introduced in the functional component conversion [where non-admin users couldn't create a new collection](https://github.com/Kinto/kinto-admin/issues/3045).
2. Corrects an issue I discovered while working on this where collection records didn't load correctly when switching collections/buckets.

### CODE REVIEW

File: src/components/collection/CollectionForm.tsx [hunk 1]
```diff
@@ -311,7 +310,7 @@ export default function CollectionForm({
         {` ${creation ? "Create" : "Update"} collection`}
       </button>
       {" or "}
-      <Link to="/">Cancel</Link>
+      <a href="/#/">Cancel</a>
```
Comment (leplatrem) [inline src/components/collection/CollectionForm.tsx:313]: Out of curiosity, why this change?

File: test/components/CollectionForm_test.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+      width: 0,
+    });
+    Range.prototype.getClientRects = () => ({
+      item: () => null,
+      length: 0,
+      [Symbol.iterator]: jest.fn(),
+    });
+  });
+
+  it("Should render an eitable form for a user with permissions creating a new collection", async () => {
```
Comment (leplatrem) [inline test/components/CollectionForm_test.js:67]: typo: editable

Comment (leplatrem) [inline test/components/CollectionForm_test.js:67]: Nitpick: maybe set the permissions here in the test setup like in other tests with `localTestProps.session.permissions.push()` to improve readability and test atomicity (leaving global `testProps` with an empty array in permissions for example

File: test/test_utils.js [hunk 3]
```diff
@@ -81,3 +82,50 @@ export function sessionFactory(props) {
     redirectURL: "",
   };
 }
+
+// temporary solution until we rewrite AdminLink to not require router around it
```
Comment (leplatrem) [inline test/test_utils.js:86]: Do we have an issue to link here for reference?

**Review by leplatrem — APPROVED** [review]
👌 
only questions/nitpicks

File: src/components/collection/CollectionForm.tsx (hunk 1 above)
Comment (alexcottner) [reply src/components/collection/CollectionForm.tsx:313]: Originally it was because we don't need a Link there to do stuff and it was messing with my unit tests. But now I have a better idea that I'm going to implement.

File: test/components/CollectionForm_test.js (hunk 2 above)
Comment (alexcottner) [reply test/components/CollectionForm_test.js:67]: Woops

File: test/test_utils.js (hunk 3 above)
Comment (alexcottner) [reply test/test_utils.js:86]: I've changed my mindset on this one a bit. Mocking Link to avoid issues with react-router is easy and I'm probably going to do that in focused unit tests going forward, so no need to remove it. 

However because we're very tied to redux right now we can't get away from Provider as easily, so there is less of a point on larger unit tests. So I created the `renderWithProvider` function that we can start migrating towards. I'm hoping this causes less friction in short and long term changes with our unit tests in this repo.

File: test/components/CollectionForm_test.js (hunk 2 above)
Comment (alexcottner) [reply test/components/CollectionForm_test.js:67]: That's a good call.

**Review by leplatrem — APPROVED** [review]
